test/e2e/explore.test.js

```diff
@@ -1,36 +1,223 @@
+        type: 'dag-pb'
+      })
+      await traverseChildren({ page, type: 'dag-pb' })
+    })
+
+    test('should open dag-cbor cid', async ({ page }) => {
+      const ipfs = kuboRpcModule.create(process.env.IPFS_RPC_ADDR)
+      const type = 'dag-cbor'
+      const cidData = new Uint8Array(Buffer.from('hello world'))
+      const dagCborAsDagJson = {
+        data: cidData
+      }
+
+      // add bytes to backend node so that explore page can load the content
+      const cidInstance = await ipfs.dag.put(dagCborAsDagJson, {
+        storeCodec: type,
+        hashAlg: 'sha2-256'
+      })
+      const cborCid = cidInstance.toString()
+
+      await testExploredCid({
+        page,
+        cid: cborCid,
+        humanReadableCID: 'base32 - cidv1 - dag-cbor - sha2-256~256~497BC2F17946B7E5DE05715EB348E47F2A6ABE6CF34ECAE9F46E236BC6E49FF5',
+        type
+      })
+      await traverseChildren({ page, type: 'dag-cbor' })
+    })
+
+    test('should open dag-pb unixFS XKCD Archives', async ({ page }) => {
+      await testExploredCid({
+        page,
+        cid: 'QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm',
+        humanReadableCID: 'base58btc - cidv0 - dag-pb - sha2-256~256~E536C7F88D731F374DCCB568AFF6F56E838A19382E488039B1CA8AD2599E82FE',
+        type: 'dag-pb'
+      })
+      await page.waitForSelector('"UnixFS"')
+      const firstChild = await page.waitForSelector('"1 - Barrel - Part 1"')
+      await firstChild.click()
+      await testExploredCid({
+        fillOutForm: false,
+        page,
+        cid: 'QmbQDovX7wRe9ek7u6QXe9zgCXkTzoUSsTFJEkrYV1HrVR',
+        humanReadableCID: 'base58btc - cidv0 - dag-pb - sha2-256~256~C212195DE60CE9B899EFDB2830101B16556018A24C7428E32198FAAB9D493F94',
+        type: 'dag-pb'
+      })
+      await page.waitForSelector('"UnixFS"')
+      await traverseChildren({ page, type: 'dag-pb' })
+    })
   })
 })
+
+/**
+ * Click the first child of the currently inspected CID and repeat until there are no more children
+ * @param {object} param0
+ * @param {import('playwright').Page} param0.page
+ * @param {string} param0.type
+ *
+ * @returns {Promise<void>}
+ */
+async function traverseChildren ({ page, type }) {
+  let hasChildren = await clickFirstExploreChild({ page, type })
+  while (hasChildren) {
+    hasChildren = await clickFirstExploreChild({ page, type })
+  }
+}
+
+/**
+ * Click the first child of the currently inspected CID if it has children
+ *
+ * - [role="rowgroup"] - table of the content, not including header
+ * - [role="row"] - each row in the table (includes headers. need to filter out by prefixing parent rowgroup selector)
+ * - [role="gridcell"] - each cell in a row
+ *
+ * @param {object} param0
+ * @param {import('playwright').Page} param0.page
+ * @param {string} param0.type
+ * @returns {Promise<boolean>} true if a child was found and clicked
+ */
+async function clickFirstExploreChild ({ page, type }) {
```

Comment:
@SgtPooki I am afraid this  is too naive –  for Apollo archives it hits the README.txt and never goes deeper:

- :green_circle: http://localhost:3000/#/explore/QmSnuWmxptJZdLJpKRarxBMS2Ju2oANVrgbr2xWbie9b2D/README.txt

We have hardcoded CIDs of examples, we need to add more explicit test than parent dir.

Just open a hardcoded, deterministic path that goes deep within directory tree.
Use :red_circle:  I mentioned.

Reply:
> (not sure if needed, but in case you see test being slow/failing on CID) we can ipfs block get specific blocks that are used in tests as static test vectors into test directory, and ipfs block put them before running E2E tests, so they work in offline mode. (you can see which blocks are needed by manually traversing dag in IPLD Explorer and looking at block requests in Network tab in devtools)

That's a useful tip. I will try that out

Reply:
Created follow up issue for this: https://github.com/ipfs/ipfs-webui/issues/2106